tests/Serial.agent.test.nut

```diff
@@ -143,11 +143,17 @@ class SerialTestCase extends ImpTestCase {
 
     function testSerialWithArrayCallFix() {
 
-        local executeCommand = Promise(function(resolve, reject) {
-            imp.wakeup(0, function() {
-                resolve("Array function called");
+        // TEST FIX TO PROMISE.SERIAL
+        // See https://forums.electricimp.com/t/my-integer-suddenly-becomes-a-string/6454/7
+
+        local executeCommand = function() {
```

Comment:
Does this actually test for the bug, i.e. does it fail if run against the unfixed version of the code? I'd expect to see either a write to a non-local variable, or an explicit assertion on type(this)...

Reply:
That seems very reasonable. Code updated. Tests fail with old lib, pass with new. BUT... this involves such an imbroglio of functions that return promises and promises themselves that I can't honestly say I'm 100% sure the test pass is due to a sensible check for success. At least it is not failing with 'indexing array with string'. Thoughts?

Reply:
Failing with the old lib, passing with the new is the acid test here I think. Yes it's a little complex, but any passer-by who is confused by it can follow the forums link, and/or review the related commits, and read all the gory details. There's already a decent comment on Promise.serial which mentions the actual issue.

Reply:
Sounds good. Thank you!